.classpath

```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
General observations:

Lack of a tool to automate software development. E.g. Maven should be added. This will simplify dependency management
Package name (validator) should be changed to comply with the convention, for example, like com.google.guava. Companies use their reversed Internet domain name to begin their package.
It's worth to extend the package structure instead of keeping everything in a flat structure. This will be useful for future project development. And it will group the classes related by responsibility.
Tests should be moved to src/test. The test class should not be called Tests.java, but should be divided into tests of individual classes.
Files:
.settings/org.eclipse.jdt.core.prefs
.project
.classpath
should not be added. Please remove it from repo. These are the settings of your local IDE.
Project is not a REST microservice. This should be changed using some framework e.g. Spring or MicroProfile/Quarkus or just JavaEE.
It is a good idea to use auto-formatting of the code, because in many places whitespace characters are missing or there are too many of them. Ctrl + Alt + L sortcut in InteliJ.
The logger was not used. Only System.out.println or System.err.println or e.printStackTrace(). The REST microservice should log to some log monitoring tool (e.g. ELK).
There are many private methods in the code that are difficult to test with unit tests. This is because the single responsibility rule is not followed. After creation separate classes with single responsibility, testing will be possible.
I've described the other detailed comments next to the lines of code they apply to.

I'm sure there are many things left that I haven't written about that could be improved. In that case, I suggest an iterative approach. Please improve what I described and when the whole project will look better, then we can do another code review and next iteration with improvements. With so many changes, there are bound to be more problems to solve.